### DISCUSSION TITLE
Error when running sunbird-ED portal server on local.

### DISCUSSION THREAD

**Discussion by Davidpavan** [Sunbird-Ed/Community#71] [Bugs]
I'm working on a local setup so that I need a portal api or token. Could you please generate and send me this mail token that would be helpful.

Unexpectedly, UWP app `37309CoolLeGetInc.CoolleMediaPlayer-ModernVLCLibra_2.1.6.0_neutral__g0y9d13zmhd68' (AUMId `37309CoolLeGetInc.CoolleMediaPlayer-ModernVLCLibra_g0y9d13zmhd68!App') supports 157 extensions but has no verbs
{"error":{"message":"Credentials must be provided when creating a service client.","stack":"Error: Credentials must be provided when creating a service client.\n    at Function.StorageServiceSettings.createFromEnvironment (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageservicesettings.js:182:9)\n    at Function.StorageServiceClient.getStorageSettings (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageserviceclient.js:925:53)\n    at new BlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\services\\blob\\blobservice.core.js:103:53)\n    at Object.exports.createBlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\azure-storage.js:98:10)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\helpers\\resourceBundleHelper.js:5:27)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\server.js:52:44)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\n    at node:internal/main/run_main_module:17:47"},"level":"error","message":"uncaughtException: Credentials must be provided when creating a service client.\nError: Credentials must be provided when creating a service client.\n    at Function.StorageServiceSettings.createFromEnvironment (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageservicesettings.js:182:9)\n    at Function.StorageServiceClient.getStorageSettings (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageserviceclient.js:925:53)\n    at new BlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\services\\blob\\blobservice.core.js:103:53)\n    at Object.exports.createBlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\azure-storage.js:98:10)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\helpers\\resourceBundleHelper.js:5:27)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\server.js:52:44)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\n    at node:internal/main/run_main_module:17:47","stack":"Error: Credentials must be provided when creating a service client.\n    at Function.StorageServiceSettings.createFromEnvironment (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageservicesettings.js:182:9)\n    at Function.StorageServiceClient.getStorageSettings (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageserviceclient.js:925:53)\n    at new BlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\services\\blob\\blobservice.core.js:103:53)\n    at Object.exports.createBlobService (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\azure-storage.js:98:10)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\helpers\\resourceBundleHelper.js:5:27)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\server.js:52:44)\n    at Module._compile (node:internal/modules/cjs/loader:1101:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)\n    at node:internal/main/run_main_module:17:47","exception":true,"date":"Wed Aug 10 2022 14:33:28 GMT+0530 (India Standard Time)","process":{"pid":13024,"uid":null,"gid":null,"cwd":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app","execPath":"C:\\Program Files\\nodejs\\node.exe","version":"v16.13.2","argv":["C:\\Program Files\\nodejs\\node.exe","C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\server.js"],"memoryUsage":{"rss":156401664,"heapTotal":108462080,"heapUsed":85164472,"external":22789889,"arrayBuffers":3393820}},"os":{"loadavg":[0,0,0],"uptime":94797},"trace":[{"column":9,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageservicesettings.js","function":"Function.StorageServiceSettings.createFromEnvironment","line":182,"method":"createFromEnvironment","native":false},{"column":53,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\common\\services\\storageserviceclient.js","function":"Function.StorageServiceClient.getStorageSettings","line":925,"method":"getStorageSettings","native":false},{"column":53,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\services\\blob\\blobservice.core.js","function":"new BlobService","line":103,"method":null,"native":false},{"column":10,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\node_modules\\azure-storage\\lib\\azure-storage.js","function":"Object.exports.createBlobService","line":98,"method":"createBlobService","native":false},{"column":27,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\helpers\\resourceBundleHelper.js","function":null,"line":5,"method":null,"native":false},{"column":14,"file":"node:internal/modules/cjs/loader","function":"Module._compile","line":1101,"method":"_compile","native":false},{"column":10,"file":"node:internal/modules/cjs/loader","function":"Module._extensions..js","line":1153,"method":".js","native":false},{"column":32,"file":"node:internal/modules/cjs/loader","function":"Module.load","line":981,"method":"load","native":false},{"column":12,"file":"node:internal/modules/cjs/loader","function":"Module._load","line":822,"method":"_load","native":false},{"column":19,"file":"node:internal/modules/cjs/loader","function":"Module.require","line":1005,"method":"require","native":false},{"column":18,"file":"node:internal/modules/cjs/helpers","function":"require","line":102,"method":null,"native":false},{"column":44,"file":"C:\\Users\\Baikuntha\\Desktop\\sunbird\\SunbirdEd-portal\\src\\app\\server.js","function":null,"line":52,"method":null,"native":false},{"column":14,"file":"node:internal/modules/cjs/loader","function":"Module._compile","line":1101,"method":"_compile","native":false},{"column":10,"file":"node:internal/modules/cjs/loader","function":"Module._extensions..js","line":1153,"method":".js","native":false},{"column":32,"file":"node:internal/modules/cjs/loader","function":"Module.load","line":981,"method":"load","native":false},{"column":12,"file":"node:internal/modules/cjs/loader","function":"Module._load","line":822,"method":"_load","native":false},{"column":12,"file":"node:internal/modules/run_main","function":"Function.executeUserEntryPoint [as runMain]","line":81,"method":"executeUserEntryPoint [as runMain]","native":false},{"column":47,"file":"node:internal/main/run_main_module","function":null,"line":17,"method":null,"native":false}],"env":"undefined.

**Comment by Davidpavan** [comment]


**Reply by Davidpavan** [reply]
Getting this error anyhelp @team

**Comment by sharathkashyap** [comment]
@Davidpavan Do you have a copy of devConfig.js in your code base in src/app/helpers folder.
cc: @rajeevsathish

**Comment by sharathkashyap** [comment]
@Davidpavan Kindly let us know if you are able to proceed further in a seperate thread in the forum. Will be locking this thread for further conversation

**Comment by gandham-santhosh** [comment]
Closing this thread as there is no response from the user
